packages/marshal/src/capDataJSON.js

```diff
@@ -0,0 +1,38 @@
+// @ts-check
+
+const { Fail } = assert;
+
+/** @param {unknown} x */
+const assertJSON = x => {
+  assert.typeof(x, 'string');
+  void JSON.parse(x);
+};
+
+/**
+ * @param {import('./types').CapData<unknown>} capData - with "simple" slots;
+ *   that is: slots whose JSON form has no occurrence of `:[`.
+ * @returns {string}
+ */
+export const capDataToJSON = ({ body, slots }) => {
+  assert(Array.isArray(slots));
+  const slotj = JSON.stringify(slots);
+  slotj.indexOf(':[') < 0 || Fail`expected simple slots`;
+  const body1 = body.replace(/^#/, '');
```

Comment:
why not check body[0] === '#' and do `body.slice(1)`, I think that's a lot more efficient.

Reply:
`body.replace(/^#/, '')` handles both smallCaps and qclass, no? (I haven't tested it, though).

Why is `.slice(1)` significantly more efficient?

Reply:
> `body.replace(/^#/, '')` handles both smallCaps and qclass, no? (I haven't tested it, though).

I guess that depends upon what this function is expected to return. Regardless of the answer to that, though, CapData like ```{ body: `{"@qclass":"bigint","digits":"0"}`, slots: [] }``` and ```{ body: `#"+0"`, slots: [] }``` represent exactly the same data (`0n`) but would have distinct `String('{"$body":{"@qclass":"bigint","digits":"0"},"slots":[]}')` and `String('{"$body":"+0","slots":[]}')` return values (respectively) from the current implementation—which seems like a problem because there's no remaining signal differentiating smallcaps from the legacy encoding.

> Why is `.slice(1)` significantly more efficient?

The answer is implementation-specific, but basically comes down to being zero-copy.
```sh
$ esbench --eshost-option '-h V8,*XS*' \
'const unprefixed="a".repeat(1000), prefixed = "#" + unprefixed' '{
  "unprefixed.replace": `result = unprefixed.replace(/^#/, "")`,
  "unprefixed.slice": `result = unprefixed.startsWith("#") ? unprefixed.slice(1) : unprefixed`,
  "prefixed.replace": `result = prefixed.replace(/^#/, "")`,
  "prefixed.slice": `result = prefixed.startsWith("#") ? prefixed.slice(1) : prefixed`,
}'
#### Moddable XS
unprefixed.replace: 0.06 ops/ms
unprefixed.slice: 3.56 ops/ms
prefixed.replace: 0.07 ops/ms
prefixed.slice: 0.95 ops/ms

#### V8
unprefixed.replace: 29.41 ops/ms
unprefixed.slice: 100.00 ops/ms
prefixed.replace: 19.61 ops/ms
prefixed.slice: 62.50 ops/ms
```

Reply:
...
> > Why is `.slice(1)` significantly more efficient?
> 
> The answer is implementation-specific, but basically comes down to being zero-copy.

I guess I trained my regex intuitions in perl where such things are optimized out the wazoo.

Thanks for the `esbench` details.